File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/07.1 - More Exercises Nested Conditional Statements/07. School Camp .py
season = input()
group_type = input()
count_students = int(input())
count_nights = int(input())

price = 0
sport = None

if season == "Winter":
    if group_type == "boys" or group_type == "girls":
        price = 9.6 * count_nights * count_students
    elif group_type == "mixed":
        price = 10 * count_nights * count_students
    if group_type == "boys":
        sport = "Judo"
    elif group_type == "girls":
        sport = "Gymnastics"
    elif group_type == "mixed":
        sport = "Ski"
elif season == "Spring":
    if group_type == "boys" or group_type == "girls":
        price = 7.2 * count_nights * count_students
    elif group_type == "mixed":
        price = 9.5 * count_nights * count_students
    if group_type == "boys":
        sport = "Tennis"
    elif group_type == "girls":
        sport = "Athletics"
    elif group_type == "mixed":
        sport = "Cycling"
elif season == "Summer":
    if group_type == "boys" or group_type == "girls":
        price = 15 * count_nights * count_students
    elif group_type == "mixed":
        price = 20 * count_nights * count_students
    if group_type == "boys":
        sport = "Football"
    elif group_type == "girls":
        sport = "Volleyball"
    elif group_type == "mixed":
        sport = "Swimming"

if count_students >= 50:
    price -= price * 0.5
elif 20 <= count_students < 50:
    price -= price * 0.15
elif 10 <= count_students < 20:
    price -= price * 0.05

print(f"{sport} {price:.2f} lv.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/10. УПРАЖНЕНИЕ ПО-СЛОЖНИ ПРОВЕРКИ/Nested Conditional Statements - More Exercises/07. School Camp.js
function schoolCamp(input) {
    let season = input.shift();  // Winter, Spring or Summer
    let groupType = input.shift();  // boys, girls or mixed
    let studentsCount = Number(input.shift());
    let nightsCount = Number(input.shift());

    let data = {
        // How it works: Season: {groupType: [priceRoom, Sport]} // Dict. -> Dict. -> List :D
        'Winter': {'boys': [9.6, 'Judo'], 'girls': [9.6, 'Gymnastics'], 'mixed': [10, 'Ski']},
        'Spring': {'boys': [7.2, 'Tennis'], 'girls': [7.2, 'Athletics'], 'mixed': [9.5, 'Cycling']},
        'Summer': {'boys': [15, 'Football'], 'girls': [15, 'Volleyball'], 'mixed': [20, 'Swimming']}
    };

    let price = data[season][groupType][0] * studentsCount * nightsCount;
    let sport = data[season][groupType][1];

    if (studentsCount >= 50) {price *= 0.5}
    if (studentsCount >= 20 && studentsCount < 50) {price *= 0.85}
    if (studentsCount >= 10 && studentsCount < 20) {price *= 0.95}

    console.log(`${sport} ${price.toFixed(2)} lv.`);
}

schoolCamp(['Spring', 'girls', '20', '7']); // Expected Output: Athletics 856.80 lv.
schoolCamp(['Winter', 'mixed', '9', '15']);  // Expected Output: Ski 1350.00 lv.
schoolCamp(['Summer', 'boys', '60', '7']);  // Expected Output: Ski 3150.00 lv.
schoolCamp(['Spring', 'mixed', '17', '14']);  // Expected Output: Ski 2147.95 lv.
